### ISSUE TITLE
Another similar block of missing content

### ISSUE THREAD

**Issue by IanMayo** [DeepBlueCLtd/LegacyMan#453]
I see we aren't picking up `PageLayer3` in `charlie_pics.html`

This is quite similar to the `anchor_pics.html` issue, though this time there isn't a `btnHist`

Again, I see it has been identified in the `Shopping list`.  It is worth throwing a warning when we can't find an element to match an item in the shopping list?

Ooh, I see the page structure for this page is quite challenging (with the anchor sitting in a relative offset from the parent PicLayer) - I guess this is the pattern you were describing earlier.

If the structure of this file introduces difficult new challenges, then I'm happy to double-check the pattern actually appears in the target data.

**Comment by robintw** [comment]
That seems to work for me - I think the fix for #451 fixes this too. Have you tried the latest version from my PR?

**Comment by robintw** [comment]
Oh hang on, I may have got confused...

**Comment by robintw** [comment]
Right, I've sorted this. It wasn't actually identified in the shopping list - the shopping list had `number2` and `number3` but not `number4`. We do raise a warning if we can't find something that's in the shopping list.

In this case, the problem was that the link to `charlie_pics.html#number4` which is in `100_unit_charlie.html` wasn't being picked up as a valid link to process and add to the shopping list. That was because that link wasn't inside a `PageLayer` div, it was inside a `layer1` div. I've changed the logic to accept links that are inside a div starting with `layer` (to get layer1, layer 2 etc) and that seems to have fixed it. I thought that might have caused a load of other errors from DITA-OT because it was picking up a load of links that it shouldn't have, but it doesn't seem to have - thank goodness.
